federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -293,11 +313,24 @@ func (frsc *ReplicaSetController) worker() {
 			return
 		}
 		key := item.(string)
-		err := frsc.reconcileReplicaSet(key)
+		status, err := frsc.reconcileReplicaSet(key)
 		frsc.replicasetWorkQueue.Done(item)
 		if err != nil {
 			glog.Errorf("Error syncing cluster controller: %v", err)
 			frsc.deliverReplicaSetByKey(key, 0, true)
+		} else {
+			switch status {
+			case statusAllOk:
+				break
+			case statusError:
+				frsc.deliverReplicaSetByKey(key, 0, true)
+			case statusNeedRecheck:
+				frsc.deliverReplicaSetByKey(key, replicaSetReviewDelay, false)
+			case statusNotSynced:
+				frsc.deliverReplicaSetByKey(key, clusterAvailableDelay, false)
+			default:
+				frsc.deliverReplicaSetByKey(key, replicaSetReviewDelay, false)
```

Comment:
Actually - when default can happen? It would mean a different status, which suggests an error. So probably you should print error here.